### ISSUE TITLE
Don't use Channels for public API

### ISSUE THREAD

**Issue by zach-klippenstein** [square/workflow#312]
While the basic Channel primitives aren't experimental anymore (`ReceiveChannel`), they are not necessarily the right abstraction to expose. We can instead use a suspending visitor/collector-style API like the new Flow library. Once Flow is stable and #311 lands, `WorkflowHost` will probably become obsolete, but until then if we can avoid exposing experimental APIs we should try to do so.

### `WorkflowHost.updates`

```kotlin
interface WorkflowHost<…> {
  val updates: ReceiveChannel<Update<…>>
}
```
can be replaced with
```kotlin
  fun collectUpdates(collector: suspend (Update<…>) -> Unit)
```

### `WorkflowHost` inputs

Not as clear what the best thing to do is. We could replicate the `Flow`/`FlowCollector` interfaces completely, but that feels a bit heavy-weight to do just for this one parameter.

### `RenderContext`

These APIs are being replaced by `Worker`s (#287).

**Comment by zach-klippenstein** [comment]
I think I am actually fine with leaving the input/output into `WorkflowHost` channels. Breaking that symmetry seems like it would be more confusing than helpful, and this isn't expected to be an end-user API anyway.

Closing as Won't Fix.
